src/widget_type.h

```diff
@@ -839,7 +839,7 @@ class Scrollbar {
 	 */
 	void ScrollTowards(size_type position)
 	{
-		if (position < this->GetPosition()) {
+		if (position <= this->GetPosition()) {
 			/* scroll up to the item */
 			this->SetPosition(position);
 		} else if (position >= this->GetPosition() + this->GetCapacity()) {
```

Comment:
I wonder whether you're solving the right issue. Basically you're saying that sometimes you get the behaviour that you request to scroll to position X and it ends up at X + 1, right?
Won't that mean that it's being run with a capacity of 0? In the old case you'd get into this branch and then set position to `position - 0 + 1`. That would match the behaviour you were seeing.